### PR TITLE
Enhancements 

### PR DESCRIPTION
**Pull request by cs-sujeet-ahirrao**
1. If playbook is active or awaiting then don't execute playbook again
2. Handled Playbook failed condition
3. Get playbook and playbook record details from $scope.payload rather than $scope.config
4.  Handle console Error: ngRepeat:dupes Duplicate Key in Repeater in [playbookPendingDecision.directive.js and playbookPendingDecision.html]

### CODE REVIEW

File: widget/view.controller.js [hunk 1]
```diff
@@ -167,19 +220,22 @@
         $limit: ALL_RECORDS_SIZE
       };
       return $resource(API.QUERY + 'tasks').save(queryString, queryBody).$promise.then(function (response) {
-        if (response['hydra:member'].length > 0 || response['hydra:member']) {
+        if (response['hydra:member'].length > 0) {
           $scope.dummyRecordIRI = response['hydra:member'][0]['@id'].replace('/api/3/', '');
```
Comment (cs-pranali-chavan) [inline widget/view.controller.js]: hydra member check has been removed

File: widget/view.controller.js [hunk 2]
```diff
@@ -167,19 +220,22 @@
...
+      }, function (error) {
+        deferred.reject(error);
       });
+      return deferred.promise;
     }
 
     function _checkDynamicVariables() {
       $resource(API.WORKFLOW + 'api/dynamic-variable/?limit=1000&name=playbook_wizard_config').get({}).$promise.then(function (response) {
-        if (response['hydra:member'].length > 0 || response['hydra:member']) {
+        if (response['hydra:member'].length > 0) {
```
Comment (cs-pranali-chavan) [inline widget/view.controller.js]: hydra member check has been removed

File: widget/view.controller.js [hunk 3]
```diff
@@ -364,15 +420,18 @@
 
     function _init() {
       if (!currentPermissionsService.availablePermission('workflows', 'execute')) {
+        toaster.error({
+          body: "You dont have permission to execute the playbooks"
```
Comment (cs-pranali-chavan) [inline widget/view.controller.js]: playbook will be singular if this code is for selected playbook

**Review by cs-pranali-chavan — CHANGES_REQUESTED** [review]
Check comments. Rest code is appropriate

**Review by cs-pranali-chavan — APPROVED** [review]
Looks appropriate

**Review by cs-pranali-chavan — CHANGES_REQUESTED** [review]
Kindly add copyright 
  MIT License
  Copyright (c) 2024 Fortinet Inc
in all html, js and css files

**Review by cs-pranali-chavan — APPROVED** [review]
Looks good. Code is appropriate

**Review by cs-suyog-jinturkar — APPROVED** [review]

**Pull request merged**